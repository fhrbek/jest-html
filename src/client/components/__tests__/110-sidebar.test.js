/* eslint-env jest */
/* eslint-disable global-require, import/newline-after-import */
import React from 'react';
import renderer from 'react-test-renderer';
import Sidebar from '../110-sidebar';

// https://github.com/facebook/react/issues/7386#issuecomment-238091398
jest.mock('react-dom');
jest.mock('react-router-dom', () => ({
  Link: require('./mockComponent')('Link'),
}));

describe('Sidebar', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Sidebar
          title="Title"
          subtitle="Subtitle"
          linkBack="link/to/parent/folder"
        >
          {['one', 'two']}
        </Sidebar>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly without linkBack', () => {
    const tree = renderer
      .create(
        <Sidebar title="Title" subtitle="Subtitle">
          {['one', 'two']}
        </Sidebar>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
