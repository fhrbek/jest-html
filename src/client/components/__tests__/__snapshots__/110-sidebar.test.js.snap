// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar renders correctly 1`] = `
<div
  id="jh-sidebar"
  style={
    Object {
      "WebkitFlex": "0 0 18em",
      "backgroundColor": "#eee",
      "borderRight": "2px solid #ddd",
      "flex": "0 0 18em",
      "fontFamily": "sans-serif",
      "fontSize": "0.8em",
      "overflow": "auto",
      "padding": "0.3em 0",
    }
  }
>
  <div
    style={
      Object {
        "alignItems": "baseline",
        "display": "flex",
        "flexDirection": "row",
        "marginBottom": "1.1em",
        "marginTop": "0.4em",
      }
    }
  >
    <div
      style={
        Object {
          "WebkitFlex": "0 0 30px",
          "flex": "0 0 30px",
        }
      }
    >
      <div
        dataMockType="Link"
        style={
          Object {
            "color": "currentColor",
            "margin": "0.3em",
            "padding": "0.3em",
          }
        }
        to="link/to/parent/folder"
      >
        <div
          style={
            Object {
              "color": "blue",
              "fontWeight": "bold",
            }
          }
        >
          Link
        </div>
        <i
          className="giu-icon fa fa-chevron-left fa-lg"
          style={
            Object {
              "color": undefined,
              "cursor": "pointer",
              "fontSize": undefined,
              "letterSpacing": "normal",
            }
          }
        />
      </div>
    </div>
    <div
      style={
        Object {
          "WebkitFlex": 1,
          "flex": 1,
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
      title="Subtitle"
    >
      Title
    </div>
    <div
      style={
        Object {
          "WebkitFlex": "0 1 30px",
          "flex": "0 1 30px",
        }
      }
    >
      <i
        className="giu-icon fa fa-file-code-o fa-lg"
        id="jh-toggle-raw"
        onClick={undefined}
        style={
          Object {
            "color": "gray",
            "cursor": "pointer",
            "fontSize": undefined,
            "letterSpacing": "normal",
          }
        }
        title="Toggle raw/HTML snapshot"
      />
    </div>
  </div>
  one
  two
</div>
------------HTML PREVIEW---------------
<div
  id="jh-sidebar"
  style="-webkit-flex:0 0 18em;background-color:#eee;border-right:2px solid #ddd;flex:0 0 18em;font-family:sans-serif;font-size:0.8em;overflow:auto;padding:0.3em 0;"
>
  <div style="align-items:baseline;display:flex;flex-direction:row;margin-bottom:1.1em;margin-top:0.4em;">
    <div style="-webkit-flex:0 0 30px;flex:0 0 30px;">
      <div
        dataMockType="Link"
        style="color:currentColor;margin:0.3em;padding:0.3em;"
        to="link/to/parent/folder"
      >
        <div style="color:blue;font-weight:bold;">
          Link
        </div>
        <i
          class="giu-icon fa fa-chevron-left fa-lg"
          style="cursor:pointer;letter-spacing:normal;"
        >
        </i>
      </div>
    </div>
    <div
      style="-webkit-flex:1;flex:1;font-weight:bold;text-align:center;"
      title="Subtitle"
    >
      Title
    </div>
    <div style="-webkit-flex:0 1 30px;flex:0 1 30px;">
      <i
        class="giu-icon fa fa-file-code-o fa-lg"
        id="jh-toggle-raw"
        style="color:gray;cursor:pointer;letter-spacing:normal;"
        title="Toggle raw/HTML snapshot"
      >
      </i>
    </div>
  </div>
  one
  two
</div>
`;

exports[`Sidebar renders correctly without linkBack 1`] = `
<div
  id="jh-sidebar"
  style={
    Object {
      "WebkitFlex": "0 0 18em",
      "backgroundColor": "#eee",
      "borderRight": "2px solid #ddd",
      "flex": "0 0 18em",
      "fontFamily": "sans-serif",
      "fontSize": "0.8em",
      "overflow": "auto",
      "padding": "0.3em 0",
    }
  }
>
  <div
    style={
      Object {
        "alignItems": "baseline",
        "display": "flex",
        "flexDirection": "row",
        "marginBottom": "1.1em",
        "marginTop": "0.4em",
      }
    }
  >
    <div
      style={
        Object {
          "WebkitFlex": "0 0 30px",
          "flex": "0 0 30px",
        }
      }
    >
      <div
        dataMockType="Link"
        style={
          Object {
            "color": "currentColor",
            "margin": "0.3em",
            "padding": "0.3em",
          }
        }
        to="#"
      >
        <div
          style={
            Object {
              "color": "blue",
              "fontWeight": "bold",
            }
          }
        >
          Link
        </div>
        <i
          className="giu-icon fa fa-chevron-left fa-lg"
          onClick={undefined}
          style={
            Object {
              "color": "#999999",
              "cursor": undefined,
              "fontSize": undefined,
              "letterSpacing": "normal",
            }
          }
        />
      </div>
    </div>
    <div
      style={
        Object {
          "WebkitFlex": 1,
          "flex": 1,
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
      title="Subtitle"
    >
      Title
    </div>
    <div
      style={
        Object {
          "WebkitFlex": "0 1 30px",
          "flex": "0 1 30px",
        }
      }
    >
      <i
        className="giu-icon fa fa-file-code-o fa-lg"
        id="jh-toggle-raw"
        onClick={undefined}
        style={
          Object {
            "color": "gray",
            "cursor": "pointer",
            "fontSize": undefined,
            "letterSpacing": "normal",
          }
        }
        title="Toggle raw/HTML snapshot"
      />
    </div>
  </div>
  one
  two
</div>
------------HTML PREVIEW---------------
<div
  id="jh-sidebar"
  style="-webkit-flex:0 0 18em;background-color:#eee;border-right:2px solid #ddd;flex:0 0 18em;font-family:sans-serif;font-size:0.8em;overflow:auto;padding:0.3em 0;"
>
  <div style="align-items:baseline;display:flex;flex-direction:row;margin-bottom:1.1em;margin-top:0.4em;">
    <div style="-webkit-flex:0 0 30px;flex:0 0 30px;">
      <div
        dataMockType="Link"
        style="color:currentColor;margin:0.3em;padding:0.3em;"
        to="#"
      >
        <div style="color:blue;font-weight:bold;">
          Link
        </div>
        <i
          class="giu-icon fa fa-chevron-left fa-lg"
          style="color:#999999;letter-spacing:normal;"
        >
        </i>
      </div>
    </div>
    <div
      style="-webkit-flex:1;flex:1;font-weight:bold;text-align:center;"
      title="Subtitle"
    >
      Title
    </div>
    <div style="-webkit-flex:0 1 30px;flex:0 1 30px;">
      <i
        class="giu-icon fa fa-file-code-o fa-lg"
        id="jh-toggle-raw"
        style="color:gray;cursor:pointer;letter-spacing:normal;"
        title="Toggle raw/HTML snapshot"
      >
      </i>
    </div>
  </div>
  one
  two
</div>
`;
